lib/Reason.ml

```diff
@@ -52,10 +52,10 @@ module Binary_reason = struct
     let rec loop = function
       | [] ->
           if is_intf_or_impl magic then
-            user_error "Unknown version" [("magic", Sexp.Atom magic)]
+            failwith (Format.sprintf "Unknown version (magic: %s)" magic)
```

Comment:
what do you think about defining `failwithf` in `Import`? (using `Format.kasprintf failwith`). I thought it was in `Base`, but it's only in `Core`.

Reply:
This would only be used twice. In the future, this code could be either gone or improved, in both cases this would no longer be useful.

Reply:
I'm fine with a dead `failwithf` in import if you think it's worth it in the meantime.

Reply:
Yes it would not be in vain, in the future we would need an easy way to emit errors, warnings and hints

Reply:
Yes, that's why I'm against adding this.
We need more than the `Failure` exception (eg. locations, colors, formatting) and I don't want that to be used elsewhere.

Reply:
I see no problem to have only what is necessary for this PR for now and rework it when we need it